Zigurat-blockchain-masters/blockchain-core
# Cryptography service implementation #19

## Pull Request

### Description

This PR includes implementation of the hashing and cryptography functions following best practices 

### Related Issue

Closes #19 
Closes #28 
Closes #29

### Changes Made

1. Implemented Hashing and crypto modules with all the functions
2. Error handling is considered too
3. Added JSDoc for every method
4. Jest set up #28 done
5. Added tests for `hashing.js` #29 


### How to Test

1. Install dependencies `npm install`
6. Make sure Unit tests are passing `npm test`


### Screenshots (if applicable)

Test execution: 

<img width="606" alt="image" src="https://github.com/Zigurat-blockchain-masters/blockchain-core/assets/32820910/2ca62150-533a-4c92-bded-bcd0ed555fea">


### By submitting this pull request, I confirm the following (please tick checkboxes):

- [x] Any new functions, methods, or classes are properly documented.
- [x] The commit messages are clear and descriptive.
- [x] The pull request title is concise and descriptive.
- [x] The pull request is targeted to the correct branch.
- [x] All unit tests have passed successfully.


### Additional Notes (if any)

Unit tests for crypto module will be implemented as part of a different PR (in order to merge this one sooner)
